pyccel/parser/semantic.py

```diff
@@ -1530,14 +1550,25 @@ def _assign_GeneratorComprehension(self, lhs_name, expr, **settings):
     #====================================================
 
 
-    def _visit(self, expr, **settings):
-        """Annotates the AST.
+    def _visit(self, expr):
+        """
+        Annotate the AST.
 
         The annotation is done by finding the appropriate function _visit_X
         for the object expr. X is the type of the object expr. If this function
         does not exist then the method resolution order is used to search for
         other compatible _visit_X functions. If none are found then an error is
-        raised
+        raised.
+        
+        Parameters
+        ----------
+        expr : object
+            Object to visit.
+        
+        Returns
+        -------
+        obj
```

Comment:
See above. Also fix the argument type. The type should be `pyccel.ast.basic.Basic`

Reply:
fixed

Reply:
Please add the commit SHA to your message when you fix things so it's easy to check that the changes fix all the comments raised.
Here I don't think the fix is correct. You have put the wrong type

Reply:
the fix was in [279c244](https://github.com/pyccel/pyccel/pull/1385/commits/279c24409e351a3ae8106efc5030a29a96156b57) do you mean the return type "pyccel.ast.variable.Variable" is wrong?

Reply:
Yes. `_visit` returns objects of lots of different types. E.g. `pyccel.ast.core.Assign` which is not a Variable. My first comment told you the correct type

Reply:
I thought you said to fix the argument's (`expr`) type.
Do you mean the return type should be `pyccel.ast.basic.Basic`
Can you please confirm from [279c244](https://github.com/pyccel/pyccel/pull/1385/commits/279c24409e351a3ae8106efc5030a29a96156b57) if the changes are right or wrong?

Reply:
Ah I understand the confusion now. Both the argument and the return object have the type `pyccel.ast.basic.Basic`. This is the type from which all elements of pyccel's AST inherit. This function takes an AST object and returns an equivalent object with more information about type etc. These two objects are equivalent but do not necessarily have the same type.

In [279c244](https://github.com/pyccel/pyccel/pull/1385/commits/279c24409e351a3ae8106efc5030a29a96156b57) I only see changes to the descriptions. I don't think it is right but this has its own conversation below

Reply:
Fixed in [0d130cb](https://github.com/pyccel/pyccel/pull/1385/commits/0d130cbf9cd3f9e77cb52f2dafeb7c15ae36e28a)